src/net.cpp

```diff
@@ -2418,7 +2418,9 @@ bool CConnman::Bind(const CService &addr, unsigned int flags, NetPermissionFlags
         return false;
     }
 
-    if (addr.IsRoutable() && fDiscover && !(flags & BF_DONT_ADVERTISE) && !(permissions & PF_NOBAN)) {
+    const bool noban{NetPermissions::HasFlag(static_cast<NetPermissionFlags>(flags), PF_NOBAN)};
```

Comment:
I stared at this for a minute to figure out why the cast added. Then I realized it's because this line uses `flags` instead of `permissions`. The names are certainly similar enough to be confusing.

Reply:
Right, it is using the wrong variable :facepalm:

Reply:
Would be nice if there was a test that failed with this diff